### PR TITLE
Fix how --null works with custom filters

### LINKED ISSUE #272: --initial-filter and --null
Combining --initial-filter and --null got wrong result.

```
#!/bin/zsh
cd /tmp
cat <<EOF > peco.bug.json
{
    "CustomFilter": {
        "grep": {
            "Cmd": "grep",
            "Args": ["$QUERY"]
        }
    }
}
EOF
echo "display\0real\n" > input.txt
# got real
peco --null --rcfile peco.bug.json --initial-filter grep input.txt
# got real
peco --null --query di input.txt
# BUG! got display
peco --null --rcfile peco.bug.json --initial-filter grep --query di input.txt
```

### PR DESCRIPTION
**Pull request by lestrrat**
fixes #272

### CODE REVIEW

**Comment by lestrrat** [comment]
あー、まてよ、これやると、ツール側がNUL付きの文字列くらっちゃうのか…

**Comment by lestrrat** [comment]
とりあえずこれでどうだ。

**Comment by yoshinari-nomura** [comment]
It seems to have been forgotton.  Something wrong with merging?

I wish I could use Migemo filter with --null option.

**Comment by lestrrat** [comment]
@yoshinari-nomura I believe the user who asked for it never replied, and therefore I never knew if this was what they wanted

**Comment by yoshinari-nomura** [comment]
Thanks. I should have checked #272 first.

IMHO, I think users, including me, are expecting Migemo to work with `--null` option without any changes.

```
"CustomFilter": {
    "Migemo": {
        "Cmd": "migemogrep",
        "Args": [ "$QUERY" ],
        "BufferThreshold": 10000
    }
}
```

That is, migemogrep simply takes non-id parts and returns matches.
This is the baseline.

As a further improvement, `ReturnsLineID` directive for null-aware filters would also be useful.

**Comment by lestrrat** [comment]
Sure.
However, [this is the current status](https://github.com/peco/peco/discussions/540) of this project. Please send us PRs if you want to see this move.

**Comment by yoshinari-nomura** [comment]
Oh, sorry for disturbing you.

**Comment by lestrrat** [comment]
@yoshinari-nomura Hope I didn't sound like "go away!" ?? That's not my intention. Let me try again:

I'm sorry, I just don't have much time and motivation, so if you want to contribute it would be certainly appreciated.

**Comment by yoshinari-nomura** [comment]
Thanks for kind words.

いえいえ，ありがとうございます．私のほうこそ冷たく聞こえたのでしたらすみません．心得ているつもりです．

golang の経験はないのですが，暇をみて貢献できればと思っています．
